### ISSUE TITLE
Linting Helm chart failed due to not conform to Kubernetes naming requirements

### ISSUE THREAD

**Issue by philwelz** [kubernetes/dashboard#5592]
##### Environment

```
Installation method: helm v3.3.4
Kubernetes version: 1.18.9
Dashboard version: 2.0.4
Operating system: Mac OS 
Node.js version ('node --version' output): v14.10.1
Go version ('go version' output): na
```

##### Steps to reproduce
1. helm lint chart/
2. helm upgrade -i -f chart/values.yaml 

##### Observed result
1. 
```
==> Linting .
[ERROR] templates/clusterrole-readonly.yaml: object name does not conform to Kubernetes naming requirements: "": invalid metadata name, must match regex ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])+$ and the length must not longer than 253
[ERROR] templates/clusterrolebinding-readonly.yaml: object name does not conform to Kubernetes naming requirements: "": invalid metadata name, must match regex ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])+$ and the length must not longer than 253
[ERROR] templates/ingress.yaml: object name does not conform to Kubernetes naming requirements: "": invalid metadata name, must match regex ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])+$ and the length must not longer than 253
[ERROR] templates/networkpolicy.yaml: object name does not conform to Kubernetes naming requirements: "": invalid metadata name, must match regex ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])+$ and the length must not longer than 253
[ERROR] templates/pdb.yaml: object name does not conform to Kubernetes naming requirements: "": invalid metadata name, must match regex ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])+$ and the length must not longer than 253
[ERROR] templates/psp.yaml: object name does not conform to Kubernetes naming requirements: "": invalid metadata name, must match regex ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])+$ and the length must not longer than 253
```

2.

```
label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; 
annotation validation error: missing key "meta.helm.sh/release-name": must be set to "kubernetes-dashboard"; 
annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "kubernetes-dashboard"
```

##### Expected result

0 Charts Failed.

```
==> Linting chart/

1 chart(s) linted, 0 chart(s) failed
```

##### Comments

Also with `fullnameOverride: 'kubernetes-dashboard'` and `nameOverride: 'kubernetes-dashboard'` set in values.yaml i have this behaviour.

**Comment by floreks** [comment]
@desaintmartin can you take a look?

**Comment by desaintmartin** [comment]
Thanks!

1. is a bug in helm 3.3.x: https://github.com/helm/helm/issues/8621. We may include all comments within the condition to, hopefully, avoid it in the meantime.
2. is interesting, could you make a PR ?

**Comment by philwelz** [comment]
Hey,

sorry for the late response. Thanks for checking this.

1. still persists, waiting for helm/helm#8621 to be merged

2. seems like it it was my coonfig issue, sorry for the trouble 😄

**Comment by philwelz** [comment]
closed bcs helm/helm#8621 was merged with helm/helm#8862 into helm relase 3.4.1
